### PR TITLE
FALCON-2258 Improve error message for definition/describe api in case extension does not exist

### PR DESCRIPTION
**Pull request by sandeepSamudrala**

### CODE REVIEW

File: prism/src/main/java/org/apache/falcon/resource/proxy/ExtensionManagerProxy.java [hunk 1]
```diff
@@ -663,7 +663,7 @@ public APIResult getExtensionDescription(
         try {
             return new APIResult(APIResult.Status.SUCCEEDED, ExtensionStore.get().getResource(extensionName, README));
         } catch (Throwable e) {
-            throw FalconWebException.newAPIException(e, Response.Status.INTERNAL_SERVER_ERROR);
+            throw FalconWebException.newAPIException(e, Response.Status.BAD_REQUEST);
```
Comment (pallavi-rao) [inline prism/src/main/java/org/apache/falcon/resource/proxy/ExtensionManagerProxy.java]: By just catching Throwable how do we know it is a bad request or a server error?

File: extensions/src/main/java/org/apache/falcon/extensions/jdbc/ExtensionMetaStore.java [hunk 2]
```diff
@@ -126,7 +126,12 @@ public ExtensionBean getDetail(String extensionName) {
         Query q = entityManager.createNamedQuery(PersistenceConstants.GET_EXTENSION);
         q.setParameter(EXTENSION_NAME, extensionName);
         try {
-            return (ExtensionBean)q.getSingleResult();
+            List resultList = q.getResultList();
+            if (resultList.size()>0) {
```
Comment (pallavi-rao) [inline extensions/src/main/java/org/apache/falcon/extensions/jdbc/ExtensionMetaStore.java]: Nit : use isEmpty instead.

File: prism/src/main/java/org/apache/falcon/resource/proxy/ExtensionManagerProxy.java (hunk 1 above)
Comment (sandeepSamudrala) [inline prism/src/main/java/org/apache/falcon/resource/proxy/ExtensionManagerProxy.java]: Made the respective changes.

File: extensions/src/main/java/org/apache/falcon/extensions/jdbc/ExtensionMetaStore.java (hunk 2 above)
Comment (sandeepSamudrala) [inline extensions/src/main/java/org/apache/falcon/extensions/jdbc/ExtensionMetaStore.java]: Ack. Made the changes.

File: prism/src/main/java/org/apache/falcon/resource/proxy/ExtensionManagerProxy.java [hunk 3]
```diff
@@ -732,7 +734,7 @@ public APIResult getExtensionDefinition(
             return new APIResult(APIResult.Status.SUCCEEDED, ExtensionStore.get().getResource(extensionName,
                     extensionName.toLowerCase() + EXTENSION_PROPERTY_JSON_SUFFIX));
         } catch (Throwable e) {
-            throw FalconWebException.newAPIException(e, Response.Status.INTERNAL_SERVER_ERROR);
+            throw FalconWebException.newAPIException(e, Response.Status.BAD_REQUEST);
```
Comment (pallavi-rao) [inline prism/src/main/java/org/apache/falcon/resource/proxy/ExtensionManagerProxy.java]: You missed changing this one.

**Pull request closed without merging**